internal/auth0/client/resource_credentials.go

```diff
@@ -197,16 +197,7 @@ func createClientCredentials(ctx context.Context, data *schema.ResourceData, met
 func readClientCredentials(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	api := meta.(*config.Config).GetAPI()
 
-	client, err := api.Client.Read(
-		ctx,
-		data.Id(),
-		management.IncludeFields(
-			"client_id",
-			"client_secret",
```

Comment:
By explicitly requiring the `client_secret` to be within the response, we were implying a requirement on the `read:client_keys` permission.